File: 1013-fibonacci-number/1013-fibonacci-number.java
class Solution {
    public int fib(int n) {
        if (n == 0) {
            return 0;
        }
        int old = 0;
        int newy = 1;
        int hold = old;

        for (int count = 1; count < n; count++) {
            //ans = old + newy;
            hold = old + newy; 
            old = newy;
            newy = hold;
        }

        return newy;
    }
}

File: 1013-fibonacci-number/1013-fibonacci-number.cpp
class Solution {
public:
    int fib(int n) {
        if (n < 2) {
            return n;
        }

        int ret = 0;
        unordered_map<int, int> dict;
        dict[0] = 0;
        dict[1] = 1;   

        for (int i = 2; i <= n; i++) {
            dict[i] = dict[i-1] + dict[i-2];
        }
        return dict[n];
    }
};